sebs/fission/fission.py

```diff
@@ -0,0 +1,373 @@
+            while True:
+                try:
+                    packageStatus = subprocess.run(
+                        f"fission package info --name {packageName}".split(),
+                        stdout=subprocess.PIPE,
+                    )
+                    subprocess.run(
+                        f"grep succeeded".split(),
+                        check=True,
+                        input=packageStatus.stdout,
+                        stderr=subprocess.DEVNULL,
+                    )
+                    break
+                except subprocess.CalledProcessError:
+                    if "failed" in packageStatus.stdout.decode("utf-8"):
+                        logging.error("Build package failed")
+                        raise Exception("Build package failed")
+                    sleep(3)
+                    continue
+            logging.info("Package ready")
+
+    def deletePackage(self, packageName: str) -> None:
+        logging.info(f"Deleting fission package...")
+        subprocess.run(f"fission package delete --name {packageName}".split())
+
+    def createFunction(self, packageName: str, name: str) -> None:
+        triggerName = f"{name}-trigger"
+        self.functionName = name
+        self.httpTriggerName = triggerName
+        logging.info(f"Deploying fission function...")
+        try:
+            triggers = subprocess.run(
+                f"fission fn list".split(), stdout=subprocess.PIPE, check=True
+            )
+            subprocess.run(
+                f"grep {name}".split(),
+                check=True,
+                input=triggers.stdout,
+                stdout=subprocess.DEVNULL,
+            )
+            logging.info(f"Function {name} already exist")
+        except subprocess.CalledProcessError:
+            subprocess.run(
+                f"fission fn create --name {name} --pkg {packageName} \
+                    --entrypoint handler.handler --env {self.envName}".split(),
+                check=True,
+            )
+        try:
+            triggers = subprocess.run(
+                f"fission httptrigger list".split(), stdout=subprocess.PIPE, check=True
+            )
+            subprocess.run(
+                f"grep {triggerName}".split(),
+                check=True,
+                input=triggers.stdout,
+                stdout=subprocess.DEVNULL,
+            )
+            logging.info(f"Trigger {triggerName} already exist")
+        except subprocess.CalledProcessError:
+            subprocess.run(
+                f"fission httptrigger create --url /benchmark --method POST \
+                --name {triggerName} --function {name}".split(),
+                check=True,
+            )
+
+    def deleteFunction(self, name: str) -> None:
+        logging.info(f"Deleting fission function...")
+        subprocess.run(f"fission fn delete --name {name}".split())
+
+    def get_function(self, code_package: Benchmark) -> Function:
```

Comment:
Take a look at OpenWhisk implementation - this looks unnecessarily verbose, and most likely this work is already done in `System`.